plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -821,25 +822,60 @@ def _render(template, context):
         return rendered
 
     @staticmethod
-    def _escape_django_template_vars(template, tag_name):
+    def _escape_django_syntax_chars(template, tag_name):
         """
-        Escape Django template variables by wrapping the specified element into `verbatim`
-        templatetag.
+        Escape Django syntax characters by replacing them with the corresponding templatetag.
+
+        NOTE: This function does not handle the following XML syntax:
+         - namespaces in the format of namespace_alias:tag_name
+         - nested tag with the same name, e.g. <a><a>...</a></a>
+         It is unlikely that the syntax above is used in the metadata of the content units.
 
         :param template: a Django template
         :type  template: basestring
         :param tag_name: name of the element to wrap
-        :type  tad_name: basestring
+        :type  tag_name: basestring
 
-        :return: a Django template with the escaped template variables in the specified element
+        :return: a Django template with the escaped syntax characters in the specified element
         :rype: basestring
         """
-        start_tag = '<%s' % tag_name
-        end_tag = '</%s>' % tag_name
-        start_templatetag = '{%% verbatim pulp_%s_escape_method %%}' % tag_name
-        end_templatetag = '{%% endverbatim pulp_%s_escape_method %%}' % tag_name
-        template = template.replace(start_tag, start_templatetag + start_tag)
-        template = template.replace(end_tag, end_tag + end_templatetag)
+        def substitute_special_chars(template):
```

Comment:
Rather than defining functions within a function, these should be private static methods in the containing namespace (I think it's `RpmBase` here).